src/scripts/tenantDataMigrations/fill-tenant-data-from-csv.js

```
@@ -0,0 +1,424 @@
+require('dotenv').config({ path: '../../.env' })
+const { Sequelize } = require('sequelize')
+const fs = require('fs')
+const path = require('path')
+const csv = require('csv-parser')
+const DatabaseConnectionManager = require('./db-connection-utils')
+
+/**
+ * CSV Data Filling Script for Chat Communications Tenant Migration
+ *
+ * This script ONLY handles data filling using CSV lookups.
+ * All schema changes (columns, indexes, foreign keys, constraints) are handled in migrations.
+ *
+ * USAGE:
+ *   node fill-tenant-data-from-csv.js
+ *
+ * REQUIREMENTS:
+ *   - DEFAULT_TENANT_CODE environment variable
+ *   - data/user_tenant_mapping.csv file (optional - if present, used for CSV-based overwrites)
+ *   - Database must be migrated first (run migrations before this script)
+ */
+
+class TenantDataFiller {
+	constructor() {
+		this.dbManager = new DatabaseConnectionManager({
+			poolMax: 10,
+			poolMin: 2,
+			logging: false,
+		})
+		this.sequelize = this.dbManager.getSequelize()
+
+		this.defaultTenantCode = process.env.DEFAULT_TENANT_CODE
+
+		if (!this.defaultTenantCode) {
+			throw new Error('DEFAULT_TENANT_CODE environment variable is required')
+		}
+
+		// CSV lookup cache for user-tenant mapping
+		this.userTenantLookupCache = new Map()
+
+		this.stats = {
+			totalProcessed: 0,
+			successfulUpdates: 0,
+			failedUpdates: 0,
+			startTime: Date.now(),
+			csvRecords: 0,
+			usersWithoutTenantCode: 0,
+			usersWithTenantCode: 0,
+		}
+
+		// Only users table for chat-communications
+		this.tableConfig = {
+			name: 'users',
+			columns: ['tenant_code'],
+			titleColumn: 'user_id',
+		}
+	}
+
+	async loadLookupData() {
+		console.log('🔄 Loading lookup data from CSV file...')
+
+		const possibleCsvPaths = [
+			path.join(__dirname, '../../../data/user_tenant_mapping.csv'),
+			path.join(__dirname, '../../data/user_tenant_mapping.csv'),
+			'/var/src/data/user_tenant_mapping.csv',
+			path.join(__dirname, '../../../src/data/user_tenant_mapping.csv'),
+		]
```

We should enforce a single, well-defined CSV path and fail fast if the file is missing.

---

@nevil-mathew Done.